### DISCUSSION TITLE
Question about architecture

### DISCUSSION THREAD

**Discussion by StokeMasterJack** [idom-team/idom#297] [Question] [answered]
What you are creating seems very exciting and useful. I just wanted to clear up a few questions of my understanding: 

Q1: These UI components are running _server-side_, correct? UI events are sent from the client to the server and then handled in the server-side components. Rendering happens on the server and then html (or dom diff) is sent to the client.Is this correct? 

Q2: Also, since the components run on the server, I can access server-side stuff (like files and databases) correct?

Q3: Assuming I'm correct on the Q1 and Q2, if I wanted to write some custom client-side code, where would that go?

Thanks.

**Comment by rmorshea** [comment, marked as answer]
Thanks for posting!

- Q1: yes
- Q2: yes
- Q3: this isn't very well documented yet, but I have a [template repository](https://github.com/idom-team/idom-react-component-cookiecutter) that can get you started. The template repo contains a `pip` installable Python package that distributes with it a custom React component. For more info on how that works refer to [the docs](http://idom-docs.herokuapp.com/docs/javascript-components.html#import-javascript-bundles).
